File: CycleChessBot/game/game_state.py
from enum import Enum

class GameState(Enum):
    ON_GOING = 0
    MATE_FOR_WHITE = 1
    MATE_FOR_BLACK = 2
    PATE = 3
    DRAW = 4
    CHECK_FOR_WHITE = 5
    CHECK_FOR_BLACK = 6

    # TODO сделать свой класс енум (отнаследовать от дефолтного и перегрузить превращение в строку)
    def __str__(self) -> str:
        return str(self.value)

File: CycleChesscpp/app/src/main/java/com/serebryakov/cyclechesscpp/application/model/game/GameState.kt
package com.serebryakov.cyclechesscpp.application.model.game

enum class GameState {
    ON_GOING,
    MATE_FOR_WHITE,
    MATE_FOR_BLACK,
    PATE,
    DRAW,
    CHECK_FOR_WHITE,
    CHECK_FOR_BLACK,
}

fun Int.toGameState(): GameState {
    return when (this) {
        0 -> GameState.ON_GOING
        1 -> GameState.MATE_FOR_WHITE
        2 -> GameState.MATE_FOR_BLACK
        3 -> GameState.PATE
        4 -> GameState.DRAW
        5 -> GameState.CHECK_FOR_WHITE
        6 -> GameState.CHECK_FOR_BLACK
        // TODO добавить сюда какой-то еще тип, чтобы понимать когда проблемы
        else -> GameState.ON_GOING
    }
}
